docs/docs/propeller/index.md

```diff
@@ -0,0 +1,36 @@
+---
+sidebar_position: 4
+---
+
+# Propeller
+
+Propeller is EquinoxJS's "[propulsion](https://github.com/jet/propulsion)
+lite". It's a set of abstractions for building reactive event processing
+pipelines catering specifically for event sourced reactions from Equinox's
+stores.
+
+## StreamsSink
+
+The `StreamsSink` receives batches of events from a source and intelligently
+manages handling them.
+
+- **Stream management**: maintains an ordered queue of incoming streams, ensuring
+  that events are processed in an orderly manner.
```

Comment:
Probably merits an individual point that the goal is to close batches and checkpoint them, and that this is achieved without needing to stop the world for the checkpointing operation to complete before we move to the next one.

```suggestion
  that reactions processing aligns with the ordering of the triggering events; the aim being to be able to checkpoint attained progress ('tick off batches') as soon as possible.
```